### PR TITLE
ATLAS : New procedure for accounting : Closes #4983

### LINKED ISSUE #4983: New procedure for accounting
Motivation
----------

In order to integrate new labels for the replicas for the site accounting :
- Custodial data has: rule with lock
- Persistent data has: rule without lifetime
- Temporary data has: rule with lifetime
- Cached data has: no rule
the Oracle accounting procedure needs to be updated

### PR DESCRIPTION
**Pull request by cserf**
ATLAS : New procedure for accounting : Closes #4983

### CODE REVIEW

File: etc/sql/oracle/schema.sql [hunk 1]
```diff
@@ -1741,7 +1742,8 @@ organization index;
 	TIER VARCHAR2(5), 
 	SPACETOKEN VARCHAR2(255), 
 	GRP_DATATYPE VARCHAR2(50), 
-	SITE VARCHAR2(255)
+	SITE VARCHAR2(255),
+        REP_TYPE NUMBER(3)
```
Comment (mlassnig) [inline etc/sql/oracle/schema.sql]: indent :-p

File: etc/sql/oracle/schema.sql [hunk 2]
```diff
@@ -1714,7 +1714,8 @@ organization index;
...
```
Comment (mlassnig) [inline etc/sql/oracle/schema.sql]: indent :-p

File: etc/sql/oracle/procedures.sql [hunk 3]
```diff
@@ -521,8 +526,28 @@ FOR i IN ( SELECT scope FROM scopes WHERE scope NOT LIKE 'mock%' AND scope NOT I
...
-        SELECT /*+ LEADING(D R RS) USE_HASH(RS) USE_HASH(R) SWAP_JOIN_INPUTS(RS) INDEX_FFS(RS ("RSES"."ID")) FULL(R) FULL(D) USE_HASH_AGGREGATION */ 
+    SITE,
+    REP_TYPE)
+        WITH l AS
+         (SELECT /*+ use_hash(a,b) full(a) full(b) */ a.scope,
+                                                      a.name,
+                                                      a.rse_id,
+                                                      MAX(a.bytes) AS fbytes,
+                                                      MAX(CASE
+                                                              WHEN b.locked=1 THEN 2
```
Comment (mlassnig) [inline etc/sql/oracle/procedures.sql:541]: shouldn't there be a fourth type for the caches? :thinking:

File: etc/sql/oracle/schema.sql (hunk 2 above)
Comment (cserf) [reply etc/sql/oracle/schema.sql]: :man_facepalming:

File: etc/sql/oracle/procedures.sql (hunk 3 above)
Comment (cserf) [reply etc/sql/oracle/procedures.sql:541]: See `NVL(l.rep_type, 4)` line 577
4 = cached
2 = custodial
1 = permanent
0 = temporary

**Review by bari12 — APPROVED** [review]
LGTM

**Review by mlassnig — APPROVED** [review]
:crossed_fingers:

**Pull request merged**